### ISSUE TITLE
Can't select Required in Taxonomies as Org Admin/User

### ISSUE THREAD

**Issue by digihash** [MISP/MISP#7341]
### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | Ubuntu
| OS version (client)       | Win10
| PHP version               | 7.3.27-1~deb10u1
| MISP version / git hash   | v2.4.141 (92f2631c44e8f86924e51ec2e7413dd8e559c08d)
| Browser                   | Chrome/Brave

### Expected behavior
As a User or Org Admin, I can see the Required column with the checkbox under Taxonomies so I expect that I can check or uncheck that checkbox without issues as it shows for my user.

### Actual behavior
When I click the checkbox in the Required column for any Taxonomy on https://mispserver/taxonomies/index, I get the following error:
![image](https://user-images.githubusercontent.com/1122246/114855505-57711600-9de6-11eb-84c5-9037fe5abea6.png)

### Steps to reproduce the behavior

1. Login with user that has User or Org Admin role (maybe other roles too), I get the error above. I guess you need to **Site Admin** permission to be able to change this flag.
2. Go to https://mispserver/taxonomies/index
3. Click the checkbox in the Required column for any taxonomy.

**Comment by chrisinmtown** [comment]
I think you're right in guessing that site admin privileges are required to toggle the state of the 'required' flag on a taxonomy.  So is this a request to improve the error message that's shown ("could not retrieve current state") to show clearly you don't have the required privileges?  Or is this a request to change the level of privileges privileges to toggle the 'required' flag?

I tried toggling a taxonomy 'required' flag on a version 2.4.133 system where I don't have admin privs and saw this message appear briefly:  'Request failed due to an unexpected error.'

**Comment by digihash** [comment]
I would say, either make the error message more clear that my user doesn't have the permission to take this action, or just remove the Required column from the view of such users. What do you think?

**Comment by iglocska** [comment]
Yeah it's an issue with the error message, or actually more the view itself, it shouldn't be a checkbox for non site admins.
